### PR TITLE
f tes mapping listTesTask method

### PR DESCRIPTION
**Pull request by evgeniimv**
implemented first version of mapping listTesTask method. Need resolve ERROR, which probably connected with filterExpression implementation.

### CODE REVIEW

**Comment by evgeniimv** [comment]
Pay Attention! 
When we use **run/filter** - or **filterRuns()** method (in case when "namePrefix" null or empty) second parameter **Boolean loadStorageLinks**, is **not supporting** in our TES. 
So, in this implementation I used **constant loadStorageLinks = true**, that maybe incorrect.

**Comment by evgeniimv** [comment]
> implemented first version of mapping listTesTask method. Need resolve ERROR, which probably connected with filterExpression implementation.

RESOLVED.

File: cloud-pipeline-common/model/src/main/java/com/epam/pipeline/vo/filter/FilterExpressionTypeVO.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+public enum FilterExpressionTypeVO {
+    LOGICAL(0),
+    AND(1),
+    OR(2);
+
+    private long id;
+    private static Map<Long, FilterExpressionTypeVO> idMap = new HashMap<>();
+
+    static {
```
Comment (SilinPavel) [inline cloud-pipeline-common/model/src/main/java/com/epam/pipeline/vo/filter/FilterExpressionTypeVO.java]: do we need this maps in common module to be able to send request?

File: cloud-pipeline-common/model/src/main/java/com/epam/pipeline/vo/filter/FilterOperandTypeVO.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    EQUALS("="),
+    NOT_EQUALS("!="),
+    MORE_OR_EQUALS(">="),
+    MORE(">");
+
+    private String operand;
+
+    private static Map<String, FilterOperandTypeVO> operandsMap = new HashMap<>();
+
+    static {
```
Comment (SilinPavel) [inline cloud-pipeline-common/model/src/main/java/com/epam/pipeline/vo/filter/FilterOperandTypeVO.java]: the same here

File: tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TesTaskServiceImpl.java [hunk 3]
```diff
@@ -57,8 +68,38 @@ public TesCreateTaskResponse submitTesTask(TesTask body) {
     }
 
     @Override
-    public TesListTasksResponse listTesTask() {
-        return new TesListTasksResponse();
+    public TesListTasksResponse listTesTask(String namePrefix, Long pageSize, String pageToken, TaskView view) {
+        TesListTasksResponse tesListTasksResponse = new TesListTasksResponse();
+        List<PipelineRun> pipelineRunList;
```
Comment (SilinPavel) [inline tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TesTaskServiceImpl.java]: try to rewrite it with `?:` operator and make it final

File: tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TesTaskServiceImpl.java [hunk 4]
```diff
@@ -57,8 +68,38 @@ public TesCreateTaskResponse submitTesTask(TesTask body) {
...
+        if (StringUtils.isNotEmpty(namePrefix)) {
+            pipelineRunList = searchRunsWithNamePrefix(namePrefix, pageSize, pageToken);
+        } else {
+            pipelineRunList = filterRunsWithOutNamePrefix(pageSize, pageToken);
+        }
+        tesListTasksResponse.setTasks(pipelineRunList.stream().map(pipelineRun ->
+                taskMapper.mapToTesTask(pipelineRun, Optional.ofNullable(view).orElse(DEFAULT_TASK_VIEW)))
+                .collect(Collectors.toList()));
+        return tesListTasksResponse;
```
Comment (SilinPavel) [inline tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TesTaskServiceImpl.java]: can we return here somethink like `new TesListTasksResponse(pipelineRunList)`? In order to get rid of tesListTasksResponse local var

File: cloud-pipeline-common/model/src/main/java/com/epam/pipeline/vo/filter/FilterExpressionTypeVO.java (hunk 1 above)
Comment (evgeniimv) [reply cloud-pipeline-common/model/src/main/java/com/epam/pipeline/vo/filter/FilterExpressionTypeVO.java]: fixed, also removed "namesMap"

File: cloud-pipeline-common/model/src/main/java/com/epam/pipeline/vo/filter/FilterOperandTypeVO.java (hunk 2 above)
Comment (evgeniimv) [reply cloud-pipeline-common/model/src/main/java/com/epam/pipeline/vo/filter/FilterOperandTypeVO.java]: fixed

File: tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TesTaskServiceImpl.java (hunk 3 above)
Comment (evgeniimv) [reply tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TesTaskServiceImpl.java]: refactored with ?: operator, removed local var

File: tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TesTaskServiceImpl.java (hunk 4 above)
Comment (evgeniimv) [reply tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TesTaskServiceImpl.java]: done

Comment (evgeniimv) [reply tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TesTaskServiceImpl.java]: Added @ NonNull annotation for first (List<TesTask> tasks) argument in TesListTasksResponse class, to be able to use constructor with only one(this) argument

File: tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TesTaskServiceImpl.java [hunk 5]
```diff
@@ -57,8 +68,32 @@ public TesCreateTaskResponse submitTesTask(TesTask body) {
     }
 
     @Override
-    public TesListTasksResponse listTesTask() {
-        return new TesListTasksResponse();
+    public TesListTasksResponse listTesTask(String namePrefix, Long pageSize, String pageToken, TaskView view) {
+        return new TesListTasksResponse((StringUtils.isNotEmpty(namePrefix) ?
```
Comment (SilinPavel) [inline tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TesTaskServiceImpl.java]: this way of writing would be easy to read
```
final listedTasks = (StringUtils.isNotEmpty(namePrefix) 
        ? searchRunsWithNamePrefix(namePrefix, pageSize, pageToken)
        : filterRunsWithOutNamePrefix(pageSize, pageToken)
).stream()
 .map(pipelineRun ->
                taskMapper.mapToTesTask(
                            pipelineRun, 
                            Optional.ofNullable(view).orElse(DEFAULT_TASK_VIEW)
                ))
 .collect(Collectors.toList())
return new TesListTasksResponse(listedTasks);
```

Comment (evgeniimv) [reply tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TesTaskServiceImpl.java]: fixed

**Review by SilinPavel — APPROVED** [review]

**Pull request merged**